applications/GeoMechanicsApplication/custom_constitutive/linear_elastic_plane_strain_2D_law.cpp

```diff
@@ -89,12 +101,83 @@ void GeoLinearElasticPlaneStrain2DLaw::CalculatePK2Stress(const Vector& rStrainV
 {
     KRATOS_TRY
 
+    UpdateInternalDeltaStrainVector(rValues.GetStrainVector());
+
     Matrix C;
     this->CalculateElasticMatrix(C, rValues);
-    // Total formulation
-    noalias(rStressVector) = prod(C, rStrainVector);
+
+    // Incremental formulation
+    noalias(mStressVector) = mStressVectorFinalized + prod(C, mDeltaStrainVector);
+
+    SetExternalStressVector(rStressVector);
 
     KRATOS_CATCH("")
 }
 
+void GeoLinearElasticPlaneStrain2DLaw::UpdateInternalDeltaStrainVector(const Vector& r_strain_vector){
+    mDeltaStrainVector = r_strain_vector - mStrainVectorFinalized;
```

Comment:
This is way clearer than before, thanks for making the change!

Reply:
Inlined the statement.